Code/JavaWrappers/Atom.i

```diff
@@ -120,4 +140,30 @@
     $self->setQuery(query);
   }
 
+  // From Python Wrapper
+  std::string AtomGetSmarts(bool doKekule=false, bool allHsExplicit=false,
+                          bool isomericSmiles=true) {
+	std::string res;
+	if (($self)->hasQuery()) {
+	  res = RDKit::SmartsWrite::GetAtomSmarts(static_cast<const RDKit::QueryAtom *>(($self)));
+	} else {
+	  // FIX: this should not be necessary
+	  res = RDKit::SmilesWrite::GetAtomSmiles(($self), doKekule, nullptr, allHsExplicit,
+									   isomericSmiles);
+	}
+	return res;
+  }
+
+  // from Python Wrapper
+  std::string describeQuery() {
+	PRECONDITION(($self), "bad atom");
```

Comment:
I don't think you need this `PRECONDITION` inside a method, we do not have it elsewhere.
```suggestion
```